### PR TITLE
Simplify servlet advice

### PR DESCRIPTION
**Pull request by laurit**

### CODE REVIEW

File: instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java
Comment (mateuszrzeszutek) [inline instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java:65]: By the way, `currentContext()` can never return null - you can simplify that bit too 😄

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java
Comment (laurit) [reply instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java]: Thanks, removed it.

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java
Comment (laurit) [inline instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java:66]: Won't using `isRecording` cause issues when running with sampling?

Comment (trask) [reply instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java:66]: ya, I think `isRecording` should only be used to short-circuit updating a given span's attributes (or name) and probably nothing else, so would support removing this check

**Review by trask — APPROVED** [review]
less code ✅ 
more comments ✅ 
tests pass ✅ 
🚀

File: instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java
Comment (laurit) [reply instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java:66]: I'll try to get rid of `isRecording` next week (my initial attempt failed on glassfish and I wasn't able to repro it locally so it requires a bit of debugging). As `isRecording` was already used before this pr I believe we can merge this if there aren't any objections.

**Pull request merged**
